### ISSUE TITLE
Not working on 0.7.2+ due to the changed lighting via shaders.

### ISSUE THREAD

**Issue by apoapostolov** [delVhariant/illumination#16]
The mod is not usable on 0.7.x builds. Both Dusk and Dawn are very dark grey, and Night is pure blackness.

**Comment by delVhariant** [comment]
Yep, I've not looked through the code yet, but from the changelog there are significant changes to lighting and color.
eg: I think my Global Illumination threshold is now going to be a built in feature, which is pretty neat.

I'll be looking at updating next week, it might take a little time depending on the extent of the changes.

**Comment by delVhariant** [comment]
@apoapostolov I've created a new branch with a version that should work on 0.7.3 (although my testing has been pretty minimal)
I'm not publishing it yet, but you should be able to install manually with the manifest url:
https://raw.githubusercontent.com/delVhariant/illumination/0.7.x-master/module.json

It's also worth mentioning that the Global Illumination threshold is now a base feature of foundry and can be set per scene in the scene config, so I've removed it from this version.

Let me know if there are any issues

**Comment by apoapostolov** [comment]
Thank you for the very timely update. I installed the 0.7.3 branch and tested it on the newly released 0.7.4. It works perfectly.

**Comment by amcvfx** [comment]
Thanks for the update! I've been testing it out and found that it partially works with 0.7.5 but doesn't do everything that the previous version could do on 0.6.6.
Without the global illumination switch in Dynamic Illumination, it doesn't seem to switch on and off correctly using the default per-scene setting. It also has no option to turn the global illumination back on once brightness passes a certain level.
Also nighttime lighting isn't being triggered by time of day like it used too. It seems to remain at dusk.

**Comment by Varriount** [comment]
@amcvfx I don't know if this is related, but sometimes the selected button for the time of day seems to get "stuck". Switching between times still works, but the GUI doesn't accurately reflect the current one.

**Comment by delVhariant** [comment]
There were a couple of bugs in the lighting in foundry 0.7.5 that were causing issues, that seem fixed in 0.7.6 and 0.7.7.
I've also attempted to fix a bug with the module that was causing the colors to get stuck if it was changed while still animating, please let me know if you're still seeing issues on the latest version of foundry/dynamic-illumination

**Comment by caerandir** [comment]
Hi delVhariant, first: Many thanks for the module!!! I have the bug with Foundry 0.7.7 that if I use color animation, if I change the dusk/dawn/day/night while another change is in progress, I cannot change anything for a while. When I switch off animation, no prob. But: the Icon in the HUD is often showing the wrong status - I switch to night, but icon in HUD is still on day, although everything else is just fine.

**Comment by delVhariant** [comment]
Hi @caerandir apologies for the lack of response.
It turns out my new buttons behaved differently from the default day/night buttons, which don't highlight at all. I'm still investigating if the highlighting/lack of highlighting changes any functionality, but for now I've made it so none of them highlight as per the default buttons.

With regards to changing during an ongoing animation, I've just pushed a new release that will by default prevent you from doing that, and provide an option to allow it that should improve the behaviour, though it's still not quite perfect with different color levels.
